homeassistant/components/tado/__init__.py

```diff
@@ -41,70 +40,48 @@
 
 CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
 
+_LOGGER = logging.getLogger(__name__)
+
 
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Set up Tado."""
 
     setup_services(hass)
-
     return True
 
 
-type TadoConfigEntry = ConfigEntry[TadoConnector]
+type TadoConfigEntry = ConfigEntry[TadoData]
 
 
 async def async_setup_entry(hass: HomeAssistant, entry: TadoConfigEntry) -> bool:
     """Set up Tado from a config entry."""
 
     _async_import_options_from_data_if_missing(hass, entry)
 
-    username = entry.data[CONF_USERNAME]
-    password = entry.data[CONF_PASSWORD]
-    fallback = entry.options.get(CONF_FALLBACK, CONST_OVERLAY_TADO_DEFAULT)
-
-    tadoconnector = TadoConnector(hass, username, password, fallback)
-
     try:
-        await hass.async_add_executor_job(tadoconnector.setup)
-    except KeyError:
-        _LOGGER.error("Failed to login to tado")
-        return False
-    except RuntimeError as exc:
-        _LOGGER.error("Failed to setup tado: %s", exc)
-        return False
-    except requests.exceptions.Timeout as ex:
-        raise ConfigEntryNotReady from ex
-    except requests.exceptions.HTTPError as ex:
-        if ex.response.status_code > 400 and ex.response.status_code < 500:
-            _LOGGER.error("Failed to login to tado: %s", ex)
-            return False
-        raise ConfigEntryNotReady from ex
-
-    # Do first update
-    await hass.async_add_executor_job(tadoconnector.update)
-
-    # Poll for updates in the background
-    entry.async_on_unload(
-        async_track_time_interval(
-            hass,
-            lambda now: tadoconnector.update(),
-            SCAN_INTERVAL,
-        )
-    )
-
-    entry.async_on_unload(
-        async_track_time_interval(
-            hass,
-            lambda now: tadoconnector.update_mobile_devices(),
-            SCAN_MOBILE_DEVICE_INTERVAL,
+        _LOGGER.debug("Setting up Tado connection")
+        tado = await hass.async_add_executor_job(
+            Tado,
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
         )
+    except PyTado.exceptions.TadoWrongCredentialsException as err:
+        raise ConfigEntryError(f"Invalid Tado credentials. Error: {err}") from err
+    except PyTado.exceptions.TadoException as err:
+        raise HomeAssistantError(f"Error during Tado setup: {err}") from err
```

Comment:
```suggestion
        raise ConfigEntryNotReady(f"Error during Tado setup: {err}") from err
```
With raising this, we will retry later